File: Logic_HandsOn/C++/Problems on bit,File Handing,Pattern etc/program130.cpp
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept one number from user and on its first 4 bits. 
//		Return modified number.
//
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

typedef unsigned int UINT;

class Solution
{
	private:

		UINT iNo;
		UINT iMask;
		UINT iResult;

	public:
		Solution(UINT iNo)
		{
			this->iNo = iNo;
			this->iMask = 0x0000000F;
                        this->iResult = 0;

		}

		UINT OffBit()
		{	

			iResult = iNo | iMask;

			return iResult;			
		}
};	

int main()
{
	UINT iValue=0;
	UINT uRet=0;

	cout<<"Enter number: ";
	cin>>iValue;

	Solution *sObj = new Solution(iValue);
	uRet = sObj->OffBit();

	cout<<"Output: "<<uRet<<"\n";

	return 0;
}

/*
OUTPUT:
Enter number: 73
Output: 79

*/

File: Logic_HandsOn/Java/Problems on bit,file handling,String etc/Program130.java
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept one number from user and on its first 4 bits. 
//		Return modified number.
//
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


import java.util.Scanner;

class Solution
{
	public int iMask,iNo;
	public int iResult;

	Solution(int iNo)
	{
		this.iNo = iNo;
		this.iMask = 0x0000000F;
		this.iResult = 0;
	}

	int OffBit()
	{

		iResult = iNo | iMask;

		return iResult;
	}	
}

class Program5
{
	public static void main(String[] args)
	{
		int iRet=0;

		Scanner sObj = new Scanner(System.in);
		
		System.out.println("Enter number: ");
		int iValue = sObj.nextInt(); 

		Solution sObj1 = new Solution(iValue);
		iRet = sObj1.OffBit();

		System.out.println("Output: "+iRet);
	}	
}
/*
OUTPUT:
Enter number: 
73
Output: 79

*/
















